### PR TITLE
Add common extra specs

### PR DESCRIPTION
**Pull request by fabi200123**
This change adds the common options:
- `disable_updates` 
- `enable_boot_debug`
- `extra_packages`
- `runner_install_template`
- `extra_context`
- `pre_install_scripts`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,7 +148,30 @@ An example of extra specs json would look like this:
...
+    }
+```
+
+*NOTE*: The `extra_context` spec adds a map of key/value pairs that are expanded in the `runner_install_template`.
```
Comment (gabriel-samfira) [inline README.md:170]:
```markdown
that **may** be expected
```

The `extra_context` spec allows you to add arbitrary key/value pairs to your custom template, which enables you to write any custom template you may need.

File: README.md [hunk 2]
```diff
@@ -115,7 +148,30 @@ An example of extra specs json would look like this:
...
+The `runner_install_template` allows us to completely override the script that installs and starts the runner. In the example above, I have added a copy of the current template from `garm-provider-common`, with the adition of one line:
+
+```bash
+echo "I am the override script: {{ .ExtraContext.Person }} - {{ .ExtraContext.Who }}" | sudo tee /garm-override.txt
```
Comment (gabriel-samfira) [inline README.md:174]: Replace this with something a bit more friently/meaningful for the example. Mine was terrible.

File: README.md [hunk 3]
```diff
@@ -115,7 +148,30 @@ An example of extra specs json would look like this:
...
+    }
+```
+
+*NOTE*: The `extra_context` spec adds a map of key/value pairs that are expanded in the `runner_install_template`.
+The `runner_install_template` allows us to completely override the script that installs and starts the runner. In the example above, I have added a copy of the current template from `garm-provider-common`, with the adition of one line:
```
Comment (gabriel-samfira) [inline README.md:171]: Add a note that this is a [golang template](https://pkg.go.dev/text/template) which enables all the goodies that come with it. Give some examples if possible.

**Review by gabriel-samfira — CHANGES_REQUESTED** [review]
Looks good, just a couple of notes.

File: README.md (hunk 2 above)
Comment (fabi200123) [reply README.md]: Updated it to do a curl to `github.com` and paste the output into a file

File: README.md (hunk 3 above)
Comment (fabi200123) [reply README.md:170]: Added a note that reference golang templates and their example regarding those

**Pull request merged**